httpe/err.go

```diff
@@ -0,0 +1,85 @@
+package httpe
+
+import (
+	"errors"
+	"net/http"
+)
+
+// StatusError wraps an http.StatusCode of value 4xx or 5xx. It is used
+// in combination with various sentinel values each representing a http
+// status code for convenience with HandlerE and HandlerFuncE.
+type StatusError int
+
+// Error returns the error message and implements the error interface.
+func (err StatusError) Error() string { return http.StatusText(int(err)) }
+
+// IsInternal return true if the error is in 5xx range. Useful for
+// error message printing and hiding details.
+func (err StatusError) IsInternal() bool { return int(err) >= 500 && int(err) <= 599 }
+
+// Code returns a status int representing and http.Status* value.
```

Comment:
s/and/an/ or is it s/and/a/ - i never knew whether to use a/an in front of haich.

Reply:
if you care to know, from what I remember from enlish class: if what follows the `a` is a vowel sound it needs to be an `an` as in an hour, everything that's an abbreviation as in http: because you say it as a-h-dge or something.
However words like house, home, honesty, humility start with a h like blowing out a candle, not a vowl and therefore only get an `a`.

done. (an)